### PR TITLE
Supports image attachment preferred content size with device screen scale

### PR DESCRIPTION
**Pull request by esreli**
In editing an image attachment preferred content size, up-scales output string with the device's main screen scale. This more accurately reflects the image scaling performed by the SDK.

### CODE REVIEW

**Comment by esreli** [comment]
Related issue: https://github.com/Esri/data-collection-ios/issues/174

File: data-collection/data-collection/Extensions/ArcGIS/AGSPopupAttachmentSize/AGSPopupAttachmentSize+String.swift [hunk 1]
```diff
@@ -31,10 +31,25 @@ extension AGSPopupAttachmentSize {
...
+    
+    var actualSizeTitle: String {
+        return titleFor(size: actualSize)
+    }
+    
+    var actualSizeScaledTitle: String {
+        return titleFor(size: actualSizeScaled)
+    }
+    
+    private func titleFor(size: CGSize) -> String {
```
Comment (philium) [inline data-collection/data-collection/Extensions/ArcGIS/AGSPopupAttachmentSize/AGSPopupAttachmentSize+String.swift]: According to the Swift API Design Guidelines, "When the first argument forms part of a prepositional phrase…the argument label should normally begin at the preposition." So `title(forSize:)` might be a more appropriate title for this method.

File: data-collection/data-collection/View Controllers/Rich Popup View Controllers/Rich Popup View Controller/RichPopupEditStagedPhotoAttachmentViewController/RichPopupEditStagedAttachmentViewController.swift [hunk 2]
```diff
@@ -185,4 +185,16 @@ class RichPopupEditStagedAttachmentViewController: UITableViewController, UIText
             return 1
         }
     }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        
+        if indexPath.section == .nameSection {
```
Comment (philium) [inline data-collection/data-collection/View Controllers/Rich Popup View Controllers/Rich Popup View Controller/RichPopupEditStagedPhotoAttachmentViewController/RichPopupEditStagedAttachmentViewController.swift]: You might consider condensing this code a little bit by retrieving the cell in only one place:

```swift
let cell = super.tableView(tableView, cellForRowAt: indexPath)
if indexPath.section == .preferredSizeSection {
    cell.textLabel?.text = AGSPopupAttachmentSize.tableViewMap[indexPath.row].actualSizeScaledTitle
}
return cell
```

File: data-collection/data-collection/Extensions/ArcGIS/AGSPopupAttachmentSize/AGSPopupAttachmentSize+String.swift [hunk 3]
```diff
@@ -31,10 +31,25 @@ extension AGSPopupAttachmentSize {
         }
     }
     
-    var asTitle: String {
+    var actualSizeScaled: CGSize {
         let size = actualSize
+        let scale = UIScreen.main.scale
+        return CGSize(width: size.width * scale, height: size.height * scale)
```
Comment (philium) [inline data-collection/data-collection/Extensions/ArcGIS/AGSPopupAttachmentSize/AGSPopupAttachmentSize+String.swift]: Another way of doing this is to apply a transform:

```swift
return size.applying(.init(scaleX: scale, y: scale))
```

**Comment by mikewilburn** [comment]
The changes introduced by this PR look good.

I confirmed on `eli.preferred-size` the following behaviors:
1. The `PREFERRED SIZE` list accurately reflects the final resolutions to which the attached images will be downsampled. Screen scaling factors are now honored).
2. Both portrait and landscape orientations are respected. 3:4 is the same as 4:3. Long edges are treated equivalently.
 i.e. a landscape image selected to be attached a resolution of 480x640 will have an uploaded resolution of 640x480, not 480x360.
3. Images that do not have the 3:4 width-to-height ratio will be scaled to ensure they fit _entirely_ into the selected resolution.
 i.e. a 16:9 image selected to be attached at a resolution of 720x960 will have an uploaded resolution of 960x540, not 720x1280.

**Pull request merged**